extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -616,29 +627,7 @@ public String getHlsUrl() throws ParsingException {
     @Override
     public StreamType getStreamType() {
         assertPageFetched();
-        return playerResponse.getObject("streamingData").has(FORMATS)
-                ? StreamType.VIDEO_STREAM : StreamType.LIVE_STREAM;
-    }
-
-    @Nullable
-    private StreamInfoItemExtractor getNextStream() throws ExtractionException {
-        try {
-            final JsonObject firstWatchNextItem = initialData.getObject("contents")
-                    .getObject("twoColumnWatchNextResults").getObject("secondaryResults")
-                    .getObject("secondaryResults").getArray("results").getObject(0);
-
-            if (!firstWatchNextItem.has("compactAutoplayRenderer")) {
-                // there is no "next" stream
-                return null;
-            }
-
-            final JsonObject videoInfo = firstWatchNextItem.getObject("compactAutoplayRenderer")
-                    .getArray("contents").getObject(0).getObject("compactVideoRenderer");
-
-            return new YoutubeStreamInfoItemExtractor(videoInfo, getTimeAgoParser());
-        } catch (final Exception e) {
-            throw new ParsingException("Could not get next video", e);
-        }
+        return streamingData.has(FORMATS) ? StreamType.VIDEO_STREAM : StreamType.LIVE_STREAM;
```

Comment:
The only problem is that for replays of livestreams, this boolean is still here so we cannot use this.

Reply:
We can check isLive if there is isLiveContent probably, as I said before isLive may mean has the live ended? -> is it a "live livestream" or livestream replay?

Am I right?

Reply:
Do you have a link for that kind of stream (live replay)? We should add a test for that case.

Reply:
I found one

https://piped.kavin.rocks/watch?v=GlPuyt_UZ_8

- it doesn't show it's live, so we are good for this point
- but it doesn't play

Reply:
I found a better way: check if `playerResponse` has a `liveStreamability` object inside `playabilityStatus` or if `isPostLiveDvr` in `playerResponse.videoDetails` is true (because we are not able to stream correctly the URLs returned (they work with segments)).